packages/router/__tests__/router-test.ts

```diff
@@ -10950,10 +10956,14 @@ describe("a router", () => {
         let rootLoaderRequest = rootLoaderStub.mock.calls[0][0]?.request;
         // @ts-expect-error
         let childLoaderRequest = childLoaderStub.mock.calls[0][0]?.request;
-        expect(rootLoaderRequest.method).toBe("GET");
+        expect(rootLoaderRequest.method).toBe("POST");
         expect(rootLoaderRequest.url).toBe("http://localhost/child");
-        expect(childLoaderRequest.method).toBe("GET");
+        expect(rootLoaderRequest.headers.get("test")).toBe("value");
+        expect(await rootLoaderRequest.text()).toBe("");
+        expect(childLoaderRequest.method).toBe("POST");
         expect(childLoaderRequest.url).toBe("http://localhost/child");
+        expect(childLoaderRequest.headers.get("test")).toBe("value");
+        // Can't re-read body here since it's the same request as the root
```

Comment:
Because `query` is a single document-level `POST` request the loaders reflect `POST` as well, but we don't proxy along the `body`